en/docs/learn/fraud-detection.md

```diff
@@ -0,0 +1,422 @@
+        // Evaluate response and decide whether to allow authentication or fail from here.
+    }
+
+    @Override
+    protected boolean retryAuthenticationEnabled() {
+        return true;
+    }
+
+    @Override
+    public String getFriendlyName() {
+        Return "Fraud Detector"
+    }
+
+    @Override
+    public boolean canHandle(HttpServletRequest httpServletRequest) {
+        Return true;
+    }
+
+    @Override
+    public String getContextIdentifier(HttpServletRequest httpServletRequest) {
+        return httpServletRequest.getParameter("sessionDataKey");
+    }
+
+    @Override
+    public String getName() {
+        return "FraudDetector";
+    }
+}
+```
+
+More information regarding how to write a custom authenticator can be found [here](https://is.docs.wso2.com/en/5.10
+.0/develop/writing-a-custom-local-authenticator/).
+
+### Intercepting User Lifecycle Management
+
+The following extension points are available in the WSO2 eventing framework to integrate with an external system and
+intercept user operations.
+
+#### Custom Event Handler
+
+WSO2 Identity Server supports the eventing framework which can be used to trigger some events such as user operations
+. Furthermore it supports pluggable handlers using which we can perform some operations based on the published events
+. Following are some examples for events.
+
+   -   PRE_AUTHENTICATION
+   -   POST_AUTHENTICATION
+   -   PRE_ADD_USER
+   -   POST_ADD_USER
+
+For integrating with a third party identity verification solution for ensuring the authenticity of the user before
+registering, we can write a custom event handler subscribed to the **PRE_ADD_USER** event.
+
+A sample event handler needs to be implemented by extending [AbstractEventHandler](https://github
+.com/wso2/carbon-identity-framework/blob/master/components/identity-event/org.wso2.carbon.identity
+.event/src/main/java/org/wso2/carbon/identity/event/handler/AbstractEventHandler.java).
+
+``` java
+Public class IdentityVerifier extends AbstractEventHandler {
+
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+
+        Map<String, Object> eventProperties = event.getEventProperties();
+        String userName = (String) eventProperties.get(IdentityEventConstants.EventProperty.USER_NAME);
+        Response response = invokeExternalIDVService(userName);
+        // Evaluate response and decide whether to allow registration or fail from here.
+    }
+
+    public String getName() {
+        return "IdentityVerifier";
+    }
+
+    @Override
+    public int getPriority(MessageContext messageContext) {
+        return 50;
+    }
+}
+```
+
+More information regarding WSO2 Eventing Framework Architecture can be found [here](https://medium
+.com/@isurakarunaratne/wso2-identity-server-eventing-framework-32505bcc1600)
+
+Some other options through which we can intercept the user operations and trigger custom flows are listed below.
```

Comment:
More options for intercepting user operations and triggering custom workflows are listed below.